test/RachisTests/RavenDB_19654.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Server.Rachis;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Logging;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace RachisTests;
+
+public class RavenDB_19654 : RachisConsensusTestBase
+{
+    public RavenDB_19654(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    private Dictionary<string, LogInfo> TestData = new Dictionary<string, LogInfo>
+    {
+        ["A"] = new LogInfo
+        {
+            CommitIndex = 664033464,
+            LastTruncatedIndex = 664033448,
+            LastTruncatedTerm = 1163367,
+            FirstEntryIndex = 664033449,
+            LastLogEntryIndex = 664033464,
+            Entries = new (long Index, long Term)[]
+            {
+                (Index: 664033449, Term: 1163368),
+                (Index: 664033454, Term: 1163370),
+            }
+        },
+        ["B"] = new LogInfo
+        {
+            CommitIndex = 664033453,
+            LastTruncatedIndex = 664033430,
+            LastTruncatedTerm = 1163367,
+            FirstEntryIndex = 664033431,
+            LastLogEntryIndex = 664033464,
+            Entries = new (long Index, long Term)[]
+            {
+                (Index: 664033431, Term: 1163367),
+                (Index: 664033449, Term: 1163368),
+            }
+        }
+    };
+    
+    [Fact]
+    public async Task CanHandleLogDivergence()
+    {
+        var leader = await CreateNetworkAndGetLeader(2);
+        var follower = GetFollowers().Single();
+
+        using (leader.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+        using (follower.ContextPool.AllocateOperationContext(out ClusterOperationContext context2))
+        using (var tx1 = context.OpenWriteTransaction())
+        using (var tx2 = context2.OpenWriteTransaction())
+        {
+            leader.CurrentLeader?.StepDown(forceElection: false);
+
+            LastCommittedTerm(leader, context, TestData["A"]);
+            LastCommittedTerm(follower, context2, TestData["B"]);
+
+            var basePath = "/home/haludi/work/ravendb/RavenDB-19654/Logs";
+             var currentDir = DateTime.Now.ToString("yyyyMMdd-hhmmss");
+             LoggingSource.Instance.SetupLogMode(LogMode.Information, Path.Combine(basePath, currentDir), TimeSpan.MaxValue, long.MaxValue, false);
```

Comment:
??

Reply:
Sorry, a leftover of debugging.
Removing this

Reply:
done